File: advent_of_code/2024/03/first.cs

using System;
using System.IO;
using System.Collections.Generic;
using System.Text.RegularExpressions;


class Program
{

    static int comp(string input)
    {
        Console.WriteLine(input);
        string substring = input.Substring(4, input.Length - 5);
        Console.WriteLine(substring);
        string[] arr = substring.Split(',');
        return int.Parse(arr[0]) * int.Parse(arr[1]);
    }
    static void Main()
    {
        string filePath = "input.txt";
        int ans = 0;

        try
        {
            string[] lines = File.ReadAllLines(filePath);

            foreach (string line in lines)
            {
                string pattern = @"mul\(\d{1,3},\d{1,3}\)";

                Regex regex = new Regex(pattern);

                MatchCollection matches = regex.Matches(line);

                foreach (Match match in matches)
                {
                    ans += comp(match.Value);
                }
            }
        }
        catch (Exception e)
        {
            Console.WriteLine(e.Message);
        }

        Console.WriteLine(ans);
    }
}

File: advent_of_code/2025/07/first.js
const fs = require('node:fs')

const input = fs.readFileSync('./input.txt', 'utf8')

const lines = input.split('\n').map(line => line.split(''))

let ans = 0

let pos = [lines[0].indexOf('S')]

for (let i = 1; i < lines.length; i++) {
  const new_pos = new Set()

  for (let j = 0; j < pos.length; j++) {
    const x = pos[j]
    if (lines[i][x] === '^') {
      new_pos.add(x - 1)
      new_pos.add(x + 1)
      ans += 1
    } else {
      new_pos.add(x)
    }
  }

  pos = Array.from(new_pos)
}

console.log(ans)